drools-wb-screens/drools-wb-scenario-simulation-editor/drools-wb-scenario-simulation-editor-client/src/main/java/org/drools/workbench/screens/scenariosimulation/client/factories/AbstractTextBoxSingletonDOMElementFactory.java

```diff
@@ -37,10 +39,19 @@ public AbstractTextBoxSingletonDOMElementFactory(final GridLienzoPanel gridPanel
 
     @Override
     public TextArea createWidget() {
-        return new TextArea() {{
-            addKeyDownHandler(DomEvent::stopPropagation);
-            addMouseDownHandler(DomEvent::stopPropagation);
-        }};
+        final TextArea textArea = GWT.create(TextArea.class);
+        textArea.addKeyDownHandler(keyDownEvent -> {
+            final int keyCode = keyDownEvent.getNativeKeyCode();
+            if (keyCode == KeyCodes.KEY_TAB) {
```

Comment:
@jomarko 
A couple of question:
1) from the ticket description it seems that somehow both TAB and ESC should be managed (in different way) but I do not see that here: are the ESC managed elsewhere or in different way?
2) is this kind of interaction specific ot ScenarioSImulation? Is it not the same for other grid editor? If the behavior should be the same on different component, then it should be better to create a builder/getter somewhere else (some abstract/parent class) to avoid duplicate this snippet

Reply:
@gitgabrio thank you for review, you had good point, the same logic was for guided decision tables.